raiden/network/transport/matrix/transport.py

```diff
@@ -677,6 +687,34 @@ def _initialize_room_inventory(self) -> None:
                 "Found room", room=room, aliases=room.aliases, members=room.get_joined_members()
             )
 
+        self._leave_unexpected_rooms(rooms_to_leave)
+
+    def _extract_partner_addresses(self, members: List[User]) -> List[Address]:
+        assert self._raiden_service is not None, "_raiden_service not set"
+        addresses = [validate_userid_signature(user) for user in members]
+        joined_partner_addresses: List[Address] = list()
+
+        return [
+            address
+            for address in addresses
+            if address not in joined_partner_addresses and self._raiden_service.address != address
+        ]
+
+    def _leave_unexpected_rooms(self, rooms_to_leave: List[Room]) -> None:
+        assert self._raiden_service is not None, "_raiden_service not set"
+
+        for room in rooms_to_leave:
+            log.warning(
+                "Leaving Room",
+                node=self._raiden_service.address,
+                reason="At least two different addresses in this room",
+                current_user=self._client.user_id,
+                room_aliases=room.aliases,
+                room_id=room.room_id,
+                partners=self._extract_partner_addresses(room.get_joined_members()),
+            )
+            room.leave()
```

Comment:
I'm a bit on the fence if we need to check the return value here. [The call can fail](https://github.com/matrix-org/matrix-python-sdk/blob/a469d531b6879c5687718937a0bfcb443155ce43/matrix_client/room.py#L386-L391) (for example if there is a short network interruption).

But since we don't add a listener to those rooms it shouldn't hurt if we are still a member.